### DISCUSSION TITLE
Got this error when uploading my build "ERROR: ERROR ITMS-90168: "null""

### DISCUSSION THREAD

**Discussion by stephanberger4** [airsdk/Adobe-Runtime-Support#2366] [Q&A]
Hi, 

I use AIR 50.1 and I want to update my IOS app but when I'm uploading with ApplicationLoader, I've got this error : 

ERROR: ERROR ITMS-90168: "null"

Any idea why ?

**Comment by ajwfrost** [comment]
That's an odd one, doesn't really help with the error details!

I'm wondering whether it may be the same issue as reported here?
https://developer.apple.com/forums/thread/701747

thanks

**Reply by ajwfrost** [reply]
Actually -> assuming it's not as simple as that (!) can you please try the "validate-app" suggestions from the below link, and we can see if that throws up any packaging issues...?

https://stackoverflow.com/questions/72006870/itmstransporter-could-not-generate-an-itmsp-null

thanks

**Reply by stephanberger4** [reply]
Thx for the answer. 
The thing is, I'm not using Xcode. I'm using Adobe Animate (for AIR / ActionScript3 code) then export a IPA build. 
With this build I'm opening ApplicationLoader on OSX and upload the build with it. 
It always worked (5 years that I do that). So I really don't understant the error...

![3](https://user-images.githubusercontent.com/6382334/208824735-124c2102-d6e4-4c94-bce8-da6cbc64b53b.png)
![1](https://user-images.githubusercontent.com/6382334/208824741-b837adb1-5492-45f2-9d79-373bf62a6493.png)
![2](https://user-images.githubusercontent.com/6382334/208824743-20b32ea7-1f09-4ee8-9a23-04ed5e9ffb38.png)

**Reply by stephanberger4** [reply]
I've tried with another app, it's working..... 
So I suppose it come from the IPA file. Need to investigate (but investigate what... THAT'S the question)

**Reply by stephanberger4** [reply]
Could it be from an ANE ?
I have a few integrate to my app but I didn't change it since 3 years... But maybe Apple change his tolerance with one of them ?

--> EDIT : Ok, it's not the ANE... I've tried to delete them all and upload the project : same error....

**Reply by ajwfrost** [reply]
Hi

There may be some issues with the code signatures I guess but a little odd to have the "null" response. Would it be possible for you to send us an IPA (easiest if it's from the build without ANEs) and we can analyse it further?
https://transfer.harman.com/requests/CiSmyIQAlJGcq906i1Gh1Z

thanks

**Reply by ajwfrost** [reply]
Thanks for that -> one weird thing, there are two entries in the IPA zip file for the launch storyboard:
```
        0  04-06-2022 18:53   Payload/appname.app/LaunchScreen.storyboardc/01J-lp-oVM-view-Ze5-6b-2t3.nib/
     1965  05-20-2020 08:32   Payload/appname.app/LaunchScreen.storyboardc/01J-lp-oVM-view-Ze5-6b-2t3.nib
```
One of them is a folder with the objects and runtime nib files, the other is just the main nib file. Apple had a version of Xcode a while ago that used this folder-based nib structure, so we'd most likely be generating that, but I can't see why the AIR developer tool would end up generating an IPA that has both of these...

Looking at the code signature, it has entries in for both of these i.e. the folder and the file are both required. But of course, when you actually unzip/expand the zip, the filesystem doesn't like the idea of having a file and a folder both with the same name.


So -> in your `InfoAdditions` section I can see a LaunchScreen storyboard being specified, but this also looks like a 'free tier' package, so you should have seen a packaging warning to say that this was going to be ignored. Our code should also then remove any existing storyboard folder that you had got in the packaging list. But I can't then work out how you've ended up with these two different storyboard folders that are causing the conflicts.. 

So just to confirm: you're using macOS for the packaging; you're running Animate with the AIR 50.1 SDK, and using the "Publish" option to create a release package..  did you see any warnings when publishing? and can you check about the storyboard, and maybe remove any of your own storyboard files and references in the app descriptor etc? We can perhaps try to confirm whether this is the problem or not, by removing the spurious files from the expanded IPA file and then re-signing it and zipping it up again (just checking, can you also run your IPA file locally on an iOS device there, or do you get an install failure for that too?)

thanks

**Reply by stephanberger4** [reply]
I'm using Windows 11 with AIR 50.1 SDK and using "Publish" option to create release package.  
Indeed I've got this error and warning "free tier version of AIR SDK will use the HARMAN launch storyboard. Conflict in file name : LaunchScreen.storyboardc/Info.plist", and the IPA can't be created. 
So I erased the "Info.plist" file in LaunchScreen.storyboardc directory. 
Doing that, the file can be created. But maybe that's the cause of the error "null". 

What can I do to not erased the info.plist and create the file without having the error "free tier version of AIR SDK will use the HARMAN launch storyboard. Conflict in file name : ...." ?
![error](https://user-images.githubusercontent.com/6382334/209250438-7e7c7c95-3e1f-477a-b306-a72686454249.png)
![folder](https://user-images.githubusercontent.com/6382334/209250469-653aaefd-d042-48c0-87fd-d6916205b908.png)

**Reply by ajwfrost** [reply]
Interesting, as we should still have code in there that detects the conflict at a folder level rather than just the files.. I'll see if we can reproduce this here to work out how to fix it properly.

In the meantime though, best is to not have that LaunchScreen.storyboardc folder in your list of files to package up, and to remove the reference to it from your application descriptor's InfoAdditions. The ADT packager will add it all in automatically.. 

thanks

**Reply by stephanberger4** [reply]
THANKS !!!! 
I've deleted the LaunchScreen.storyboardc folder and removed the reference  from my application decriptor and it worked ! It uploaded the build !!!

Thank you VERY MUCH !

Note : 
I've received this email from Apple : 

Dear Developer,We identified one or more issues with a recent delivery for your app, "Marées-NC" 8.5.0 (8.5.0). Your delivery was successful, but you may wish to correct the following issues in your next delivery:ITMS-90683: Missing purpose string in Info.plist - Your app’s code references one or more APIs that access sensitive user data, or the app has one or more entitlements that permit such access. The Info.plist file for the “Marees NC.app” bundle should contain a NSLocationAlwaysAndWhenInUseUsageDescription key with a user-facing purpose string explaining clearly and completely why your app needs the data. If you’re using external libraries or SDKs, they may reference APIs that require a purpose string. While your app might not use these APIs, a purpose string is still required. For details, visit: https://developer.apple.com/documentation/uikit/protecting_the_user_s_privacy/requesting_access_to_protected_resources.After you’ve corrected the issues, you can upload a new binary to App Store Connect.Best regards,The App Store Team

**Reply by stephanberger4** [reply]
(Note that it always display this message but it doesn't matter for now, as the build is still uploaded : 
![message](https://user-images.githubusercontent.com/6382334/209299825-b7cd358d-2797-479c-a61e-72dd53ccfacb.png) )
